src/overlays/actors/ovl_En_Test6/z_en_test6.c

```diff
@@ -1094,228 +1173,245 @@ void func_80A92950(EnTest6* this, PlayState* play) {
         }
     }
 
-    if (this->unk_276 == 1) {
-        for (i = 0; i < ARRAY_COUNT(D_80A94910); i++) {
-            D_80A94910[i].unk_08 += 2.0f * ((2.0f * Rand_ZeroOne()) - 1.0f);
-            D_80A94910[i].unk_10 += 2.0f * ((2.0f * Rand_ZeroOne()) - 1.0f);
-            D_80A94910[i].unk_0C += 3.0f;
+    if (this->drawType == SOTCS_DRAW_RESET_CYCLE_SOT) {
+        for (i = 0; i < ARRAY_COUNT(sSoTCsAmmoDrops); i++) {
+            sSoTCsAmmoDrops[i].pos.x += 2.0f * ((2.0f * Rand_ZeroOne()) - 1.0f);
+            sSoTCsAmmoDrops[i].pos.z += 2.0f * ((2.0f * Rand_ZeroOne()) - 1.0f);
+            sSoTCsAmmoDrops[i].pos.y += 3.0f;
 
-            if (player->actor.world.pos.y < D_80A94910[i].unk_0C) {
-                temp_f0 = D_80A94910[i].unk_0C - player->actor.world.pos.y;
-                if (temp_f0 > 550.0f) {
-                    temp_f0 = 1.0f;
+            if (player->actor.world.pos.y < sSoTCsAmmoDrops[i].pos.y) {
+                yDiff = sSoTCsAmmoDrops[i].pos.y - player->actor.world.pos.y;
+                if (yDiff > 550.0f) {
+                    yDiff = 1.0f;
                 } else {
-                    temp_f0 = temp_f0 / 550.0f;
+                    yDiff /= 550.0f;
                 }
-                D_80A94910[i].unk_04 = SQ(temp_f0);
+                sSoTCsAmmoDrops[i].scale = SQ(yDiff);
             } else {
-                D_80A94910[i].unk_04 = -10.0f;
+                sSoTCsAmmoDrops[i].scale = -10.0f;
             }
         }
     }
-    this->unk_278++;
+    this->counter++;
 }
 
-void func_80A93298(EnTest6* this, PlayState* play) {
-    s16 temp_s3;
-    s16 temp_s4;
-    f32 phi_f24;
-    s16 phi_s2;
+/**
+ * Draws clocks in a double spiral above and below player
+ */
+void EnTest6_DrawThreeDayResetSoTCutscene(EnTest6* this, PlayState* play) {
+    s16 clock1Yaw;
+    s16 clock2Yaw;
+    s16 angle;
     s32 i;
-    f32 cos;
-    f32 sin;
+    Vec3f clockPos;
 
     OPEN_DISPS(play->state.gfxCtx);
 
-    this->unk_148 = POLY_OPA_DISP;
-    phi_f24 = 0.0f;
-
-    temp_s3 = this->unk_27C;
-    temp_s4 = (s32)(Math_SinS(play->state.frames) * 12000.0f) + temp_s3 + 0x4E20;
-    phi_s2 = (play->state.frames & 0x3C) * 1024;
-    phi_s2 *= (this->unk_154 / 200.0f);
-    this->unk_27C += (s16)this->unk_154;
-    this->unk_27E = (s16)((this->unk_154 / 200.0f) * 256.0f);
-
-    for (i = 0; i < ARRAY_COUNT(this->unk_254[0]); i++) {
-        temp_s3 += 0x1000;
-        cos = Math_CosS(temp_s3) * this->unk_150;
-        sin = Math_SinS(temp_s3) * this->unk_150;
-        Matrix_Translate(cos, phi_f24, sin, MTXMODE_NEW);
+    this->gfx = POLY_OPA_DISP;
+    clockPos.y = 0.0f;
+
+    clock1Yaw = this->clockAngle;
+    clock2Yaw = clock1Yaw + 0x4E20 + (s32)(0x2EE0 * Math_SinS(play->state.frames));
+    angle = (play->state.frames & 0x3C) * 1024;
+    angle *= this->clockSpeed / 200.0f;
+    this->clockAngle += (s16)this->clockSpeed;
+    this->clockRingRotZ = (s16)((this->clockSpeed / 200.0f) * 256.0f);
```

Comment:
Those are some whacky maths monkaS

Reply:
Oh yeah, I worked out that the `& 0x3C` ensures the `angle` updates only once every 4 frames. I made that comment elsewhere but will add that in here